File: leetcode/704 Binary Search/main.java
// Solved by @kitanoyoru
// https://leetcode.com/problems/binary-search/

class Solution {
   public static int search(int[] nums, int target) {
      int start = 0, end = nums.length - 1;

      while (start <= end) {
         int mid = (int) Math.floor((start + end) / 2);

         if (nums[mid] == target) {
            return mid;
         } else if (nums[mid] < target) {
            start += 1;
         } else {
            end -= 1;
         }
      }

      return -1;
   }
}

File: leetcode/704 Binary Search/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/binary-search/

const search = (nums: number[], target: number): number => {
  let start = 0,
    end = nums.length - 1

  while (start <= end) {
    const mid = Math.floor((start + end) / 2)

    if (nums[mid] === target) {
      return mid
    } else if (target < nums[mid]) {
      end -= 1
    } else {
      start += 1
    }
  }
  return -1
}
